examples/src/main/java/com/mapbox/navigation/examples/core/MapboxNavigationActivity.kt

```diff
@@ -156,6 +158,23 @@ class MapboxNavigationActivity : AppCompatActivity() {
             }
         }
 
+    private val maneuverCallback = object : ManeuverCallback {
+        override fun onError(error: Expected<ManeuverError, Maneuver>) {
+        }
+
+        override fun onManeuvers(maneuvers: Expected<ManeuverError, List<Maneuver>>) {
+            binding.maneuverView.renderManeuver(maneuvers)
+            maneuvers.fold(
+                { error ->
+                    //
+                },
+                { list ->
+                    binding.maneuverView.renderUpcomingManeuvers(list.subList(1, list.size))
```

Comment:
I don't like the approach where if you want to show/hide the upcoming maneuver, you control the visibility modifier. If, give a longer route, you don't want to render the upcoming maneuvers, with the approach of handling using modifier, the `View` would always render these list of complex instructions on every route progress event, even though the end user doesn't want it. Exposing a separate API makes it more clear and gives you more control whether you want to render the upcoming maneuvers altogether. In this example I add this line of code because I want to render the upcoming maneuvers. You can choose not to do this if you don't want to. Step distance remaining is a different component because it is a part of current maneuver. When you render a current maneuver, you would render all the data associated with it rather than partially. I would like to leave it as is.
cc @LukasPaczos